ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -760,6 +760,41 @@ void seal() throws StoreException {
     persistBloomFilter();
   }
 
+  /**
+   * @return index value of last PUT record in this index segment. Return {@code null} if no PUT is found
+   */
+  IndexValue getIndexValueOfLastPut() throws StoreException {
+    IndexValue indexValueOfLastPut = null;
+    if (sealed.get()) {
+      ByteBuffer readBuf = serEntries.duplicate();
```

Comment:
Is duplicate() needed?

Reply:
solved. duplicate() create a bytebuffer which shares the same content but different limit, position and capacity.